Added useful details on Matrix vs. Telegram and Matrix vs. Discord (#10)
austinhuang0131/joinmatrix: guide/matrix-vs-telegram.md

| Feature | Telegram | Matrix |
| ------- | -------- | ------ |

> | **Registration** | Requires phone number. | **Phone number is usually optional.** Depending on homeserver, **email may not be required**. There is usually no human check after registration. |

Reviewer: See above.

Reviewer 2: SSO and 3PID makes it possible to enforce Phone numbers on matrix tho its almost purely theoretical at this point. The point is more that its completely possible. 

```suggestion
| **Registration** | Requires phone number. | **Phone number is usually optional.** Depending on homeserver, **email is commonly required**. There is usually no human check after registration. |
```

| Group chat privacy | You may deny new members from reading more than 100 messages prior to them joining. | You may deny new members from reading messages prior to them being invited / joining. You may also allow or deny guest access (such as [Matrix Static](http://view.matrix.org/)) from reading messages. You may also enable encryption[^1]. |
| Publicity | Any group or channel set to public can be listed in search results, but how they are shown is arbitrary, as global search is not always visible to users. | Each homeserver has a room directory which anyone in that homeserver may publish to. |
| Invite | Through directly inviting users, or through generating invite links. | Through directly inviting users, or through shareable [addresses](../features/#promotion). |
| Permissions | Permissions of each administrator are set manually. All admins are equal (except owner). | 2^54 power levels (I think it's -2^53 to 2^53-1, however I highly doubt you will *ever* reach that limit). A user acquires a permission if their power level is equal to or higher than the power level required for the specific permission. |

> | Ads | Popular channels now carry ads that you cannot opt out. | **Most Matrix home servers do not allow ads in the chats hosted in them. No known Matrix home server inserts ads into room state.** |

Reviewer: I guess it is possible for a homeserver to specifically implement ads, but then users can still opt out pretty easily (eg. ACL). Servers can also manipulate events to remove ads. So I guess it is not possible to *effectively* implement ads?

Author: There are actually more sinister ways such as appearing as another room member from your home server and sending ads as them, while not routing that ad-injection message to the user impersonated (a home server admin might easily impersonate a local user with the current event signing scheme).

Reviewer: So homeservers doing that can be ACL'd?

Reviewer 2: I think the point is more that matrix is not immune to this attack and yes ofc you can just ACL ban those homeservers from your room to make it so a homeserver you dont agree with is ACL banned if your in control of a room.

Reviewer: ```suggestion
| Ads | Popular channels now carry ads that you cannot opt out. | It is technically possible for a homeserver to insert ads, but **there are no known occurrences**. |
```

| Invite | Through directly inviting users, or through generating invite links. | Through directly inviting users, or through shareable [addresses](../features/#promotion). |
| Permissions | Permissions of each administrator are set manually. All admins are equal (except owner). Permissions of a member do not survive leave and rejoin. | 2^54 power levels (I think it's -2^53 to 2^53-1, however I highly doubt you will *ever* reach that limit). A user acquires a permission if their power level is equal to or higher than the power level required for the specific permission. Power levels of members survive leave and rejoin. |
| Disabled and deleted account handling | Disabling an account is reversible until one year after disabling. Messages from deleted accounts survive for one more year from deletion. | Disabling an account is usually irreversible. Messages from disabled accounts are not sent to further users and servers. Rooms created by disabled accounts stay. |
| Ads | Popular channels now carry ads that you cannot opt out. | It is technically possible for a homeserver to insert ads, but **there are no known occurrences**. |

> | **Network access** | **IPv4 supported, IPv6 broken**.[^3]  | **IPv4&v6 support varies depending on homeserver**. |

Reviewer 2: ```suggestion
| **Network access** | **IPv4 supported, IPv6 broken**.[^3]  | ***Most if not all homeservers participating in the public federation have IPv4 connectivity but IPv6 connectivity varies from homeserver to homeserver.**. |
```

| 2FA | One-time token sent to another session. | Not required for login, but required (QR code, emoji verification, or Security Key) for viewing past encrypted messages. |
| **Text messages** | Maximum 4096 characters. Supports Markdown. | **Up to ~65200 bytes (up to ~40770 bytes if a formatted message with plain text fallback sent).[^2] [Supports Markdown and HTML.](../features/#text)** |
| Attachments | **Maximum 2 GB.** | Maximum 50~100 MB (for most homeservers; customizable if you run your own homeserver). |
| Reactions | Very limited. Must be enabled by group admins in groups. | All unicode emotes and [text](../features/#reactions). |
| Stickers | Up to 200 packs of 120 static or 50 animated each. | **Unlimited (static or animated) with setup.** See [here](../features/#stickers). |
| Public read receipts | Supported ambiguously. | **Supported.** |
| **Direct messages** | Not encrypted unless explicitly opted into secret chat, which cannot be carried across devices. VoIP is encrypted. | **Encrypted by default**, including VoIP. |
| **Group chats** | You can join up to 500 groups and channels. | You can join an **unlimited** amount of rooms. |

> | VoIP in groups | Supported. | Not supported until early 2022. |

Reviewer 2: Technically supported but only via Jitsi Widgets and Element clients are the only clients that have proper support for this. Expected to be replaced during 2022 with a better system that is a part of the matrix spec. 
```suggestion
| VoIP in groups | Supported. | Limited Support via Integration with Jitsi. Expected to be replaced by a better solution during 2022. |
```

| Group chat permissions | Permissions of each administrator are set manually. All admins are equal (except owner). Permissions of a member do not survive leave and rejoin. | 2^54 power levels (I think it's -2^53 to 2^53-1, however I highly doubt you will *ever* reach that limit). A user acquires a permission if their power level is equal to or higher than the power level required for the specific permission. Power levels of members survive leave and rejoin. |
| Size limits of group chats | Up to 100k members in groups, unlimited in one-to-many channels. | No artificial limits, albeit current implementations do not perform well past a few tens of thousands of members. | 
| Disabled and deleted account handling | Disabling an account is reversible until one year after disabling. Messages from deleted accounts survive for one more year from deletion. | Disabling an account is usually irreversible. Messages from disabled accounts are not sent to further users and servers. Rooms created by disabled accounts stay. |
| Ads | Popular channels now carry ads that you cannot opt out. | It is technically possible for a homeserver to insert ads, but **there are no known occurrences**. |

> | **Network access** | **IPv4 supported, IPv6 broken**.[^3]  | ***Most if not all homeservers participating in the public federation have IPv4 connectivity but IPv6 connectivity varies from homeserver to homeserver.**. |

Reviewer: ```suggestion
| **Network access** | **IPv4 supported, [IPv6 broken](https://flameeyes.blog/2017/08/06/ipv6-horror-story-telegram/)**.  | ***Most if not all homeservers participating in the public federation have IPv4 connectivity but IPv6 connectivity varies from homeserver to homeserver.**. |
```
